opengate/engines.py

```diff
@@ -291,6 +368,68 @@ def create_g4_optical_properties_table(material_properties_dictionary):
     return g4_material_table
 
 
+def create_g4_surface_properties(surface_properties_table, surface_name):
+    """
+    This function passes the surface properties stored and creates
+    Geant4 objects required to create an optical surface.
+    """
+    g4_surface = g4.G4OpticalSurface(g4.G4String(surface_name))
+
+    # Set Model
+    model = surface_properties_table["base_properties"]["surface_model"]
+    model_enum = getattr(g4.G4OpticalSurfaceModel, model, None)
+
+    if model_enum is not None:
+        g4_surface.SetModel(model_enum)
+    else:
+        fatal("Model in not present in SurfaceProperties.xml")
+
+    # Set Type
+    surface_type = surface_properties_table["base_properties"]["surface_type"]
```

Comment:
as above: I would prefer surface_type_name and surface_type
The _enum suffix is misleading because it suggests it's an enum, while it is just extracted from an enum.